### ISSUE TITLE
Custom markers are no longer case-insensitive

### ISSUE THREAD

**Issue by JeroenDeDauw** [JeroenDeDauw/Maps#491]
Reported by @acnetj in https://github.com/JeroenDeDauw/Maps/issues/488

**Comment by JeroenDeDauw** [comment]
@acnetj please specify the exact version of the Maps extension you are using and what custom markers you are talking about (ie is this in display_map or ask queries). A minimal wikitext example would be nice.

**Comment by acnetj** [comment]
Semantic Maps (Version 2.0.1.2)	
Semantic MediaWiki (Version 1.8)
External Data (Version 1.8.2)
Maps (Version 2.0.1 RC1)

Old version of map with 2 custom icons: 
![image](https://user-images.githubusercontent.com/10605369/48603547-aa032b00-e92b-11e8-940e-7b3e59520c7d.png)

See markers starting with "File:" is not working:
https://www.transit.wiki/Sandbox_2

**Comment by JeroenDeDauw** [comment]
> See markers starting with "File:" is not working:
https://www.transit.wiki/Sandbox_2

There are no markers used in the wikitext there?

**Comment by JeroenDeDauw** [comment]
@acnetj please post a *minimal* wikitext to reproduce this issue. At present I do not know where to look.

**Comment by acnetj** [comment]
Actually to clarify, the File: can only be used in icon parameter, and does not work in a separate icon format. See this page: https://www.transit.wiki/Sandbox_2

Two points with same custom icon: 
{{#display_map: 37.775110, -122.419320~yes;37.765202, -122.419688
 |service=leaflet|layers=HERE.Street|height=250px|icon=File:Rail20.png
}}

One of the points with a custom icon, but File: does not work: 
{{#display_map: 37.775110, -122.419320~yes~ ~File:rail20.png;37.765202, -122.419688
 |service=leaflet|layers=HERE.Street|height=250px
}}

One of the points with a custom icon, OK if it is the URL to the file:
{{#display_map: 37.775110, -122.419320~yes~ ~/images/8/88/Rail20.png;37.765202, -122.419688
 |service=leaflet|layers=HERE.Street|height=250px
}}

The latter style of code can be generated by the Map Editor.
The older version of Maps/Semantic Maps allowed me to integrate this with ask or compound query, and use File: for custom icon. I can't show it now because I can't make maps with external data due to tailing ";" issue.

{{#compound_query:[[latlong::34.050355, -118.254706 (2 mile)||33.990024, -118.464647 (2 mile)]][[Category:Rail stations]];icon=Rail20.png
    |[[latlong::34.050355, -118.254706 (2 mile)||33.990024, -118.464647 (2 mile)||33.754413, -118.282883 (2 mile)]];?latlong;icon=mm_20_red.png;limit=100
    |format=googlemaps|width=90%|height=100%|center=33.928361, -118.285286|zoom=10|locations={{#for_external_table:
{{{latitude}}},{{{longitude}}}~{{{stationname}}}~Metro Bike Share hub~File:Bshareicon.png;
}}}}

**Comment by JeroenDeDauw** [comment]
You can put wikitext in code blocks, like this:

![image](https://user-images.githubusercontent.com/146040/48722720-fb116880-ec24-11e8-82a2-617812fbfcaa.png)

**Comment by JeroenDeDauw** [comment]
There was an issue with this parameter some versions of Maps ago, though it was fixed. If you use Maps 5.6 or later it should work. You can see it work here 

https://sandbox.semantic-mediawiki.org/wiki/Maps_issue_491

Please update the sandbox page I linked in a way that reproduces the issue.

**Comment by acnetj** [comment]
OK so far I guess it checked out. The File: file name format should also be case sensitive (first letter be capitalized). I was upgrading the wiki from 2.0 a couple of weeks ago to version 5.0 before I installed PHP 7. I separated the upgrade process in two pieces since there were to many things to be done in one step. I would need to take another step to upgrade MW to 1.31 and Maps to 6 after upgrading PHP 7.

**Comment by JeroenDeDauw** [comment]
It seems to me you are saying there is some issue with capitalization. Your description is ambiguous though. Please create an example on the sandbox page that should work but does not due to the issue you are talking about.

**Comment by acnetj** [comment]
Well I didn't know it had to do with case sensitivity. The File: works with 5.6 but the marker links should always be case sensitive.

This is what I found upon further examination:

The 2.0 version the file name wasn't case sensitive. 
icon=mm_20_red.png (shown under v 2.0, not now)
icon=Mm_20_red.png (OK)
icon=File:mm_20_red.png (shown under v 2.0, not now)
icon=File:Mm_20_red.png (OK)

The "File" does not have to be case sensitive, so "file:" is still OK.

What I was concerned with originally was that the maps that appeared OK suddenly don't look properly. When things wasn't case sensitive then I produced the code without the expectation of case sensitivity and then copy that code to templates and many more pages. Some file names was done through copy and paste so upper case got preserved. Others file were named without regard to case and then enter into the code by hand without assuming upper case of the first letter.
